tests/Unit/Utils/UtilsTest.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Unit Tests: Util Functions
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Unit;
+
+use Yoast\WPTestUtils\BrainMonkey\TestCase;
+
+use const Parsely\Utils\DATE_UTC_FORMAT;
+
+use function Parsely\Utils\convert_to_associative_array;
+use function Parsely\Utils\get_formatted_number;
+use function Parsely\Utils\get_formatted_time;
+use function Parsely\Utils\get_utc_date_format;
+
+/**
+ * Unit Tests: Util Functions
+ *
+ * @phpstan-type Test_UTC_Date_Format_Data array{
+ *   args: array{
+ *     days: int,
+ *   },
+ *   expected_output: string,
+ *   msg: string,
+ * }
+ *
+ * @phpstan-type Test_Formatted_Number_Data array{
+ *   args: array{
+ *     number: int,
+ *   },
+ *   expected_output: string,
+ *   msg: string,
+ * }
+ *
+ * @phpstan-type Test_Formatted_Time_Data array{
+ *   args: array{
+ *     seconds: int,
+ *   },
+ *   expected_output: string,
+ *   msg: string,
+ * }
+ *
+ * @phpstan-type Test_Convert_To_Associative_Data array{
+ *   args: array{
+ *     obj: stdClass,
+ *   },
+ *   expected_output: array<mixed>,
+ *   msg: string,
+ * }
+ */
+final class UtilsTest extends TestCase {
+	/**
+	 * Test get_utc_date_format function.
+	 *
+	 * @covers function \Parsely\Utils\get_utc_date_format
+	 */
+	public function test_get_utc_date_format(): void {
+		$current_date = gmdate( DATE_UTC_FORMAT );
+		$past_date    = gmdate( DATE_UTC_FORMAT, strtotime( '-1 days' ) );
+		$future_date  = gmdate( DATE_UTC_FORMAT, strtotime( '1 days' ) );
+
+		/**
+		 * Variable.
+		 *
+		 * @var array<Test_UTC_Date_Format_Data>
+		 */
+		$tests_data = array(
+			array(
+				'args'            => array( 'days' => 0 ),
+				'expected_output' => $current_date,
+				'msg'             => 'Should match with the format of the current date.',
+			),
+			array(
+				'args'            => array( 'days' => -1 ),
+				'expected_output' => $past_date,
+				'msg'             => 'Should match with the format of the past date.',
+			),
+			array(
+				'args'            => array( 'days' => 1 ),
+				'expected_output' => $future_date,
+				'msg'             => 'Should match with the format of the future date.',
+			),
+		);
+
+		foreach ( $tests_data as $t ) {
+			$args   = $t['args'];
+			$output = get_utc_date_format( $args['days'] );
+
+			self::assertEquals( $t['expected_output'], $output, $t['msg'] );
+		}
```

Comment:
Instead of using multiple asserts using loop to verify different arguments (FYI: This pattern is very common in GoLang)